client/asset/eth/eth.go

```diff
@@ -670,12 +691,155 @@ func (eth *ExchangeWallet) unlockFunds(coins asset.Coins) error {
+
+	eth.lockedFundsMtx.Lock()
+	defer eth.lockedFundsMtx.Unlock()
+
+	var totalInputValue uint64
+	for _, input := range swaps.Inputs {
+		if _, ok := eth.lockedFunds[input.ID().String()]; !ok {
+			return nil, nil, 0,
+				fmt.Errorf("Swap: attempting to use coin that was not locked")
+		}
+		totalInputValue += input.Value()
+	}
+
+	var totalContractValue uint64
+	secretHashMap := make(map[string]bool)
+	for _, contract := range swaps.Contracts {
+		totalContractValue += contract.Value
+
+		if len(contract.SecretHash) != 32 {
+			return nil, nil, 0,
+				fmt.Errorf("Swap: expected secret hash of length 32, but got %v", len(contract.SecretHash))
+		}
+		encSecretHash := hex.EncodeToString(contract.SecretHash)
+		if _, ok := secretHashMap[encSecretHash]; ok {
+			return nil, nil, 0, fmt.Errorf("Swap: secret hashes must be unique")
+		}
+		secretHashMap[encSecretHash] = true
+		var secretHash [32]byte
+		copy(secretHash[:], contract.SecretHash)
+
+		if !common.IsHexAddress(contract.Address) {
+			return nil, nil, 0, fmt.Errorf("Swap: address in contract is not valid %v", contract.Address)
+		}
+		address := common.HexToAddress(contract.Address)
+
+		initiations = append(initiations,
+			dexeth.ETHSwapInitiation{
+				RefundTimestamp: big.NewInt(0).SetUint64(contract.LockTime),
+				SecretHash:      secretHash,
+				Participant:     address,
+				Value:           big.NewInt(0).SetUint64(contract.Value),
+			})
+	}
+
+	initGas, err := eth.getInitGas(len(initiations))
+	if err != nil {
+		return nil, nil, 0, err
+	}
+
+	totalUsedValue := totalContractValue + (initGas * swaps.FeeRate)
+	if totalInputValue < totalUsedValue {
+		return nil, nil, 0, fmt.Errorf("Swap: coin inputs value %d < required %d", totalInputValue, totalUsedValue)
+	}
+
+	opts := bind.TransactOpts{
+		From:      eth.acct.Address,
+		Value:     big.NewInt(0).SetUint64(totalContractValue),
+		GasFeeCap: big.NewInt(0).SetUint64(swaps.FeeRate),
+		GasTipCap: big.NewInt(GasTipCap),
+		Context:   eth.ctx}
+
+	tx, err := eth.node.initiate(&opts, eth.networkID, initiations)
+	if err != nil {
+		return nil, nil, 0, fmt.Errorf("Swap: initiate error: %w", err)
+	}
+
+	var txID [32]byte
+	copy(txID[:], tx.Hash().Bytes())
+	for i, initiation := range initiations {
+		txCoinID := &srveth.TxCoinID{
+			TxID:  txID,
+			Index: uint32(i)}
+		swapCoinID := srveth.SwapCoinID{
+			ContractAddress: mainnetContractAddr,
+			SecretHash:      initiation.SecretHash}
+		receipts = append(receipts,
+			&swapReceipt{
+				expiration: time.Unix(initiation.RefundTimestamp.Int64(), 0),
+				value:      initiation.Value.Uint64(),
+				txCoinID:   *txCoinID,
+				swapCoinID: swapCoinID,
+			})
+	}
+
+	eth.unlockFunds(swaps.Inputs)
+	var change asset.Coin
+	changeAmount := totalInputValue - totalUsedValue
+	if changeAmount > 0 {
+		change = eth.createAmountCoin(changeAmount)
+	}
+	if swaps.LockChange && change != nil {
+		eth.lockFunds(asset.Coins{change})
+	}
+
+	feesUsed := swaps.FeeRate * initGas
```

Comment:
We have `initGas * swaps.FeeRate` above when calculating `totalUsedValue`, so how about we assign it to a variable like `maxPossibleFee` to convey it's meaning better in both places.